### PR TITLE
Add CloseIndexResponse to HLRC

### PR DESCRIPTION
**Pull request by tlrx**
We improved the `CloseIndexResponse` in #39687; this pull request exposes it in the HLRC. It also improves the test coverage by using `AbstractWireSerializingTestCase` and `AbstractResponseTestCase`.

### CODE REVIEW

**Review by dnhatn — COMMENTED** [review]
This looks good. I left a comment to discuss. Thanks @tlrx.

File: server/src/main/java/org/elasticsearch/action/admin/indices/close/CloseIndexResponse.java [hunk 1]
```diff
@@ -86,8 +108,42 @@ public String toString() {
...
+    public static CloseIndexResponse fromXContent(final XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
     public static class IndexResult implements Writeable, ToXContentFragment {
 
+        @SuppressWarnings("unchecked")
+        private static final ConstructingObjectParser<IndexResult, String> PARSER = new ConstructingObjectParser<>("index_result", true,
+            (args, name) -> {
+                final Index index = new Index(name, "_na_");
```
[the remark this answers is not in the document]
Comment (tlrx) [reply server/src/main/java/org/elasticsearch/action/admin/indices/close/CloseIndexResponse.java]: I thought that we kind of tolerate things like this in the HLRC but my knowledge was just outdated. I talked to @hub-cap about this and he confirmed that we should instead use dedicated HLRC objects.

Thanks for spotting this issue @dnhatn

**Comment by tlrx** [comment]
@dnhatn I reverted my changes and I created dedicated objects for the HLRC close index response. This duplicates some logic (mostly in tests) but I think it's cleaner now.

Can you please have another look?

**Comment by tlrx** [comment]
@elasticmachine update branch

**Comment by dnhatn** [comment]
@elasticmachine update branch

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks for the extra iteration.

**Comment by tlrx** [comment]
Thanks a lot @dnhatn !

**Pull request merged**